### PR TITLE
WL-4007 Adding exception handling …

### PR DESCRIPTION
**Pull request by surya-janani**
prevent stack trace getting displayed to the end user

### CODE REVIEW

File: impl/src/java/org/sakaiproject/poll/service/impl/PollVoteManagerImpl.java [hunk 1]
```diff
@@ -137,11 +140,16 @@ public boolean userHasVoted(Long pollid, String userID) {
...
-		if (votes.size() > 0)
-			return true;
-		else
+		try {
+			List<Vote> votes = dao.findBySearch(Vote.class, search);
+			if (votes.size() > 0)
+				return true;
+			else
+				return false;
+		}catch(Exception e) {
```
Comment (buckett) [inline impl/src/java/org/sakaiproject/poll/service/impl/PollVoteManagerImpl.java:149]: It's bad form todo `catch (Exception e)` with something like this, it means that if you have something like a `NullPointerException` it's really hard to track down.

We should really be expecting Exceptions like this in the normal code path.

**Comment by buckett** [comment]
I'm not sure this is the correct fix as the calling code should be doing a check rather than dealing with the null  user ID in this way.

**Comment by buckett** [comment]
I've reverted this in the sakai-10.x branch as we're going to do a release soon.

**Comment by surya-janani** [comment]
1. I was trying to catch exception as I thought  we may hit other wierd database exceptions in doing dao.findBySearch. 
2. I did some research on best practise for exception handling. I feel your comment is based on throw early and catch later principle. But I don't really understand what difference it would make-,by throwing exception here(userHasVoted method) and catching the same in calling method, which is just one step below in the call stack.

Will it be better if I do a null check and skip dao.findBySearch() when its null.

**Comment by buckett** [comment]
It's generally much better to see the error, have it be reported and us to understand what is going wrong. Catching all exceptions is generally only used where you really never want the thread to fail.

It's very easy to just surround all code with try/catch blocks which ends up with errors being masked unless you are working with a debugger. If you expect something to be null the code is much clearer if it does an explicit check for it.

**Comment by surya-janani** [comment]
Okies..I get the point

**Pull request merged**